### PR TITLE
added match_environment to api request

### PR DESCRIPTION
**Pull request by CharlieCurious**
**What does this PR do?**

   Adds match env to general player stats request.

**Database changed?**

   No

**Where should the reviewer start?**

**How should this be manually tested?**

  run the app and ask for rankings

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-126](https://aptoide.atlassian.net/browse/ESKL-126)

**Questions:**

**Code Review Checklist**

- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java [hunk 1]
```diff
@@ -34,13 +34,16 @@
 
   public static final String USER_ID = "USER_ID";
   public static final String WALLET_ADDRESS = "WALLET_ADDRESS";
+  public static final String SKU = "SKU";
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java:37]: this is not being used

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java [hunk 2]
```diff
@@ -34,13 +34,16 @@
...
+  public static final String MATCH_ENVIRONMENT = "MATCH_ENVIRONMENT";
   public static final String SESSION = "SESSION";
   public static final String LOCAL_GAME_STATUS = "LOCAL_GAME_STATUS";
 
   private static final String ENTRY_PRICE_DUEL = "1 USD";
   private static final String ENTRY_PRICE_MULTIPLAYER = "4 USD";
 
   private final String userId = "string_user_id";
+  private MatchDetails.Environment match_environment;
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java:46]: for naming variables in java or kotlin, you should use camelCase. you can follow this document regarding variables by oracle: https://docs.oracle.com/javase/tutorial/java/nutsandbolts/variables.html

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java [hunk 3]
```diff
@@ -92,6 +96,7 @@ private void resumeGame(LocalGameStatus localGameStatus) {
   }
 
   private void showCreateTicket(MatchDetails.Environment environment) {
+    this.match_environment = environment;
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java:99]: usually `this` is only used when you want to distinguish class variables from input method arguments. in this case, you could just simply `matchEnvironment = environment;`

File: 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java [hunk 4]
```diff
@@ -33,10 +35,11 @@
   @Inject UserDetailsHelper userDetailsHelper;
 
   public static Intent newIntent(Context context, String userId, String walletAddress,
-      String session, LocalGameStatus localGameStatus) {
+      MatchDetails.Environment match_environment, String session, LocalGameStatus localGameStatus) {
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java:38]: naming convention also applies here

File: 2048/2048/src/main/java/com/appcoins/eskills2048/api/GeneralPlayerStats.java [hunk 5]
```diff
@@ -11,6 +11,7 @@
   @Headers("accept: application/json") @GET("room/statistics/general_player_stats")
   Single<GeneralPlayerStatsResponse> getGeneralPlayerStats(
       @Query("package_name") String packageName, @Query("wallet_address") String walletAddress,
-      @Query("ranks_above_user") int ranksAbove, @Query("ranks_below_user") int ranksBelow,
-      @Query("time_frame") String timeframe);
+      @Query("product") String product, @Query("ranks_above_user") int ranksAbove,
+      @Query("ranks_below_user") int ranksBelow,
+      @Query("match_environment") String matchEnvironment, @Query("time_frame") String timeframe);
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/api/GeneralPlayerStats.java:16]: you don't need to convert this to String. you can just use Environment type and let retrofit deal with the conversion (see SetFinalScoreUseCase)

**Review by nnelas — CHANGES_REQUESTED** [review]

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java (hunk 1 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java]: fixed

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java (hunk 2 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java]: fixed

File: 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java (hunk 3 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/LaunchActivity.java]: fixed

File: 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java (hunk 4 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java]: fixed

File: 2048/2048/src/main/java/com/appcoins/eskills2048/api/GeneralPlayerStats.java (hunk 5 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/api/GeneralPlayerStats.java]: fixed

File: 2048/2048/src/main/java/com/appcoins/eskills2048/repository/StatisticsRepository.java [hunk 6]
```diff
@@ -14,7 +17,9 @@
   }
 
   public Single<GeneralPlayerStatsResponse> getUserStatistics(String applicationId,
-      String userWalletAddress, StatisticsTimeFrame timeFrame) {
-    return api.getGeneralPlayerStats(applicationId, userWalletAddress, 3, 2, timeFrame.name());
+      String userWalletAddress, MatchDetails.Environment matchEnvironment,
+      StatisticsTimeFrame timeFrame) {
+    return api.getGeneralPlayerStats(applicationId, userWalletAddress, null, 3, 2, matchEnvironment,
```
Comment (nnelas) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/repository/StatisticsRepository.java:22]: wait... why are you sending a null? shouldn't `product` be optional on the server side?

**Review by nnelas — CHANGES_REQUESTED** [review]

File: 2048/2048/src/main/java/com/appcoins/eskills2048/repository/StatisticsRepository.java (hunk 6 above)
Comment (CharlieCurious) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/repository/StatisticsRepository.java:22]: It is, on the server side. It is not on the method getGeneralPlayerStats. I simply abstained from changing the method's signature. Should I remove the 'product' parameter?

**Review by nnelas — APPROVED** [review]

**Pull request merged**